u8timeseries/models/statistics.py

```diff
@@ -0,0 +1,136 @@
+import matplotlib.pyplot as plt
+from typing import Tuple
+import math
+
+
+def check_seasonality(ts: 'TimeSeries', m: int = None, max_lag: int = 24, alpha: float = 5):
+    """
+    Returns whether the TimeSeries [ts] is seasonal with period [m] or not.
+
+    If [m] is None, the seasonality period is inferred from the Auto-correlation Function (ACF).
+
+    :param ts: The TimeSeries to check for seasonality.
+    :param m: The seasonality period to check.
+    :param max_lag: The maximal lag allowed in the ACF.
+    :param alpha: The desired confidence level (default 5%).
+    :return: A tuple (season, m), where s is a boolean indicating whether the TimeSeries has seasonality or not
+             and m is the seasonality period.
+    """
+    if m is not None:
+        if m < 0:
+            raise ValueError('m must be a positive integer.')
+
+    n_unique = np.unique(ts.values()).shape[0]
+
+    if n_unique > 1:  # Check for non-constant TimeSeries
+
+        r = acf(ts.values(), nlags=max_lag)  # In case user wants to check for seasonality higher than 24 steps.
+
+        grad = np.gradient(r)
+        signs_changes = np.diff(np.sign(grad))
+
+        if m is None:
+            # Tries to infer seasonality from AutoCorrelation Function if no value of m has been provided.
+            # We look for the first positive local maximum of the ACF by checking the sign changes in the gradient.
+
+            # Local maximum is indicated by signs_change == -2
+            if len(np.nonzero((signs_changes == -2))[0]) > 0:
+                m = np.nonzero((signs_changes == -2))[0][0] + 1
+            else:
+                return False, 0
+
+        # Check for local maximum when m is user defined
+        if np.nonzero((signs_changes == -2))[0][0] != m-1:
+            return False, m
+
+        # Remove r[0], the auto-correlation at order 0, that introduces bias.
+        r = r[1:]
+
+        stat = bartlett_formula(r, m - 1, len(ts))
+
+        # The upper limit of the significance interval.
+        band_upper = r.mean() + norm.ppf(1 - alpha/200)*r.var()
+
+        # Only local maximum with positive values are candidates to be tested against significance level.
+        season = r[m-1] > stat * band_upper
+
+        return season, m
+
+    return False, 0
+
+
+def bartlett_formula(r, m, length):
+    """
+    Computes the standard error of [r] at order [m] with respect to [length] according to Bartlett's formula.
+
+    :param r: The array whose standard error is to be computed.
+    :param m: The order of the standard error.
+    :param length: The size of the underlying sample to be used.
+    :return: The standard error of [r] with order [m].
+    """
+    if m == 1:
+        return math.sqrt(1/length)
+    else:
+        return math.sqrt((1 + 2 * sum(map(lambda x: x ** 2, r[:m-1]))) / length)
+
+
+def seasonal_adjustment(ts: 'TimeSeries', frequency, model='multiplicative') -> Tuple['TimeSeries', 'TimeSeries']:
```

Comment:
How about breaking this method in two:
```
def extract_seasonality_and_trend(ts: 'TimeSeries', frequency) -> Tuple['TimeSeries', 'TimeSeries']
```
and
```
def remove_seasonality(ts: 'TimeSeries', frequency, model='multiplicative') -> 'TimeSeries'
```
The second one would call the first, but first one might also be useful for users. In addition the second one can return only one time series (a new TimeSeries with seasonality removed).

Reply:
Broke it down into 2 pieces (extraction and seasonality removal), and added trend removal as a third piece.